nav2_costmap_2d/plugins/static_layer.cpp

```diff
@@ -411,8 +411,17 @@ StaticLayer::updateCosts(
     return;
   }
 
+  std::vector<std::pair<MapLocation, unsigned char>> map_region_to_restore;
   if (footprint_clearing_enabled_) {
-    setConvexPolygonCost(transformed_footprint_, nav2_costmap_2d::FREE_SPACE);
+    std::vector<MapLocation> polygon_cells;
+    getCellsOccupiedByPolygon(transformed_footprint_, polygon_cells);
```

Comment:
I think this getter should also return the costs so that you don't need to then have the next step here be grabbing the costs.

The return could be the vector of pairs (or map locations and costs).